lua/nvim-treesitter/parsers.lua

```diff
@@ -244,6 +244,15 @@ list.php = {
   maintainers = { "@tk-shirasaka" },
 }
 
+list.phpdoc = {
+  install_info = {
+    url = "https://github.com/claytonrcarter/tree-sitter-phpdoc",
+    files = { "src/parser.c" },
+  },
+  used_by = { "php" },
```

Comment:
This line breaks PHP parsing completely. And if I try to open playground I get:

```
Error executing vim.schedule lua callback: ...local/share/nvim/runtime/lua/vim/treesitter/language.lua:33: ABI version mismatch for /home/mike/.local/share/nvim/plugge
d/nvim-treesitter/parser/phpdoc.so: supported between 13 and 13, found 12                                                                                              
stack traceback:                                                                                                                                                       
        [C]: in function '_ts_add_language'                                                                                                                            
        ...local/share/nvim/runtime/lua/vim/treesitter/language.lua:33: in function 'require_language'                                                                 
        /usr/local/share/nvim/runtime/lua/vim/treesitter.lua:38: in function '_create_parser'                                                                          
        /usr/local/share/nvim/runtime/lua/vim/treesitter.lua:93: in function 'get_parser'                                                                              
        ...d/playground/lua/nvim-treesitter-playground/internal.lua:50: in function 'get_node_at_cursor'                                                               
        ...d/playground/lua/nvim-treesitter-playground/internal.lua:413: in function 'fn'                                                                              
        ...gged/playground/lua/nvim-treesitter-playground/utils.lua:23: in function 'cb'                                                                               
        vim.lua:285: in function <vim.lua:285>
```

If I remove the line, PHP works again - but comments are still not parsed. At least I can't see anything in playground. I only see a single `comment` node. Pressing `t` for injected languages also doesn't help. 

Any ideas?

Reply:
Ok, created a PR with a re-generated `parser.c` from the latest TS version. That fixed the issue for me. Let's see if the PR gets merged soon. If not I'll create a fork that we can maybe include here.

https://github.com/claytonrcarter/tree-sitter-phpdoc/pull/2

Reply:
You probably want to inject phpdoc into PHP using PHP's injections.scm. A `used_by` is not necessary.

Reply:
Right, then I'll drop this line. The injection file is already included in this PR.

There's another issue with the phpdoc parser though: The author can't upgrade to TS 0.19 as this would break the integration into Atom. If you have an idea how to solve this maybe you can comment there? https://github.com/claytonrcarter/tree-sitter-phpdoc/pull/2#issuecomment-1014682817

Reply:
Add `    requires_generate_from_grammar = true,` as some parsers already have it. If the parser requires npm for generation then also `    generate_requires_npm = true,`

Reply:
Not sure if I understand. The parser does not require npm or generation from grammar. It includes a `parser.c` ready for compilation.

The issue is rather that the author wants to keep a version for TS 0.17 in his repo. So we will probably use 2 branches and we will reference a `revision` here (the one compatible with TS 0.2x).

Reply:
Oh, I think I see: If we generate from grammer it will always use the correct version of treesitter to build the `parser.c`. I'll check.

Reply:
It will be the correct version when the user has the correct version of tree-sitter and `./node_modules/.bin` not in PATH or else it will use the old version of tree-sitter.

Reply:
Pushed the change. It works locally, but some tests here seem to not like it.

Could it have to do with the version constraint in the phpdoc's `package.json`?

https://github.com/claytonrcarter/tree-sitter-phpdoc/blob/0864e30e70a53f751596657f75b14dd042cd4793/package.json#L28-L29

Reply:
Sorry, I forgot to actually push the changes. Tests also seem fine now - problem solved. Thank you!